core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/JsonProtocolMarshaller.java

```diff
@@ -97,9 +97,10 @@ private static JsonMarshallerRegistry createMarshallerRegistry() {
         return JsonMarshallerRegistry
             .builder()
             .payloadMarshaller(MarshallingType.STRING, SimpleTypeJsonMarshaller.STRING)
-            .payloadMarshaller(MarshallingType.INTEGER, SimpleTypeJsonMarshaller.INTEGER)
             .payloadMarshaller(MarshallingType.LONG, SimpleTypeJsonMarshaller.LONG)
+            .payloadMarshaller(MarshallingType.INTEGER, SimpleTypeJsonMarshaller.INTEGER)
```

Comment:
Seems like this was unintentinal. could we undo this ?

Reply:
It's on purpose, the idea being ordering the integer types by size: `long < int < short < byte`. Let me know if you feel strong about it.

Reply:
Changed, all the other places were missing as pointed out by Ani.